### PR TITLE
 add bulk action endpoint

### PR DESCRIPTION
**Pull request by ZouhourArrow**

### CODE REVIEW

File: src/licenses/licensesClient.ts [hunk 1]
```diff
@@ -356,6 +363,8 @@ export type LicensePricingRate = {
   applyOnNextBillingPeriod: boolean;
 };
 
+//export type BulkArguments = BulkBodyArgument;
```
Comment (fdianoux) [inline src/licenses/licensesClient.ts:366]: delete plz

File: src/licenses/types/bulkArguments.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  BulkBodyFields.SPECIAL_PRICE_RATE_TYPE &
+    BulkBodyFields.SPECIAL_PRICE_RATE_VALUE &
+    BulkBodyFields.SPECIAL_RATE_EFFECTIVE_APPLICATION_DATE
+>;
+
+export type BulkSetRateBody = Except<
+  BulkBodyArgument,
+  BulkBodyFields.AUTO_RENEW_STATUS
+>;
+/*export type BulkAutoRenewBody = {
```
Comment (fdianoux) [inline src/licenses/types/bulkArguments.ts:47]: why

**Review by sabirab2 — APPROVED** [review]

**Review by Dhalsim404 — APPROVED** [review]

File: src/abstractRestfulClient.ts [hunk 3]
```diff
@@ -470,6 +470,10 @@ export abstract class AbstractRestfulClient extends AbstractHttpClient {
       ] = `${this.perPage}`;
     }
 
+    // to keep client stateless
+    this.page = 1;
```
Comment (BlondJP) [inline src/abstractRestfulClient.ts:474]: +1

**Review by BlondJP — APPROVED** [review]
ok for me

**Pull request merged**